workshops/fabric-e2e-rag/workshop.md

```diff
@@ -0,0 +1,480 @@
+
+## Prerequisites
+
+To follow along with this work begin by Right-click or `Ctrl + Click` on the button below to open the Azure Portal in a new tab and begin deployment of the necessary resources.
+
+[![Deploy to Azure](https://aka.ms/deploytoazurebutton)](https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2Fmicrosoft%2moaw%2Fmain.json)
+
+This deploys the following resources:
+1. Azure OpenAI Service
+2. Azure AI Search
+3. Azure AI Services
+4. Azure Key Vault
+
+<div class="info" data-title="Note">
+  
+  > The Azure Open AI deployment also includes the deployment of the ***gpt-4*** and ***text-embedding-ada-002 models*** 
+</div>
+
+---
+
+# Loading and Preprocessing PDF Documents
+In this section, we'll load the PDF document into a spark dataframe then read the documents using the Azure AI Document Intelligence in Azure AI Services. 
+
+After extracting the text from the PDF documents, we'll use SynapseML to split the documents into chunks. The document is split into chunks to allow for more granular representation and processing of the document content.
+
+To begin with [create a Lakehouse](https://learn.microsoft.com/en-us/fabric/data-engineering/tutorial-build-lakehouse#create-a-lakehouse?WT.mc_id=data-114676-jndemenge) in your Microsoft Fabric workspace. Next in the Data Engineering workload create a new Notebook to continue.
+
+We'll then need to install the necessary libraries. We'll use the [semantic-kernel](https://learn.microsoft.com/en-us/semantic-kernel/overview?WT.mc_id=data-114676-jndemenge) library to build the RAG application with models from Azure OpenAI. 
+
+Copy the following code into a new cell and run it to install the libraries.
+
+```python
+%pip install --quiet semantic-kernel
+```
+Next we'll need to provide the keys for Azure AI Services and Azure OpenAI to access the respective services.Copy the values from the Azure Portal and paste them into the following code cell.
+
+
+```python
+
+# Azure Open AI
+aoai_service_name = ''
+aoai_endpoint = f'https://{aoai_service_name}.openai.azure.com/'
+aoai_key = ''
+aoai_deployment_name_embeddings = "text-embedding-ada-002"
+aoai_deployment_name_completions = "gpt-4"
+
+# Azure AI Search
+aisearch_name = ''
+aisearch_index_name = 'wwireports'
+aisearch_api_key = ''
+
+# Azure AI Service
+ai_services_key = ''
+ai_services_location = ''
+```
+Upload the file you'd like to analyze to the `Files` folder in your Lakehouse.
+
+Now that we have the basic configuration to Azure AI Services and Azure OpenAI, we can start loading the PDF documents into a Spark DataFrame using the `spark.read.format("binaryFile")`method provided by Apache Spark. Change the `document_path` variable to the path of the PDF document you'd like to analyze.
+
+```python
+# Import required pyspark libraries
+
+from pyspark.sql.functions import udf
+from pyspark.sql.types import StringType
+
+document_path = 'Files/your_filename.pdf' # Path to the PDF document 
```

Comment:
We should provide a demo PDF file. We have some samples here: https://github.com/Azure-Samples/azure-openai-rag-workshop/tree/main/data